RoboticonColony/Assets/Code/Roboticon.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// A Roboticon. Holds all information about the roboticon and facilitates upgrading said roboticon. 
+/// </summary>
+sealed public class Roboticon {
+
+    /// <summary>
+    /// The current Tile that the roboticon is assigned to. 
+    /// Also allows reference to itself to read, not write. 
+    /// </summary>
+    public Tile CurrentTile { get; private set; }
+
+    /// <summary>
+    /// _bonusProduction is a dictionary which holds the int value (Total production of each recource) it produces.
+    /// </summary>
+    private Dictionary<ItemType, int> _bonusProduction;
+
+    /// <summary>
+    /// CurrentCustomisations is a list of RoboticonCustomisations which the roboticon currently has applied. 
+    /// Also allows reference to itself to read, not write. 
+    /// </summary>
+    public List<RoboticonCustomisation> CurrentCustomisations { get; private set; }
+
+    /// <summary>
+    /// Creates a Roboticon instance that creates a non-customised roboticon in the specified tile.
+    /// </summary>
+    /// <param name="selectedTile"> The Tile that the roboticon is being placed on. </param>
+    public Roboticon(Tile selectedTile)
+    {
+        CurrentTile = selectedTile;
+        _bonusProduction = new Dictionary<ItemType, int>();
+        CurrentCustomisations = new List<RoboticonCustomisation>();
+
+        // Set Bonus Production to base roboticon stats
+        _bonusProduction[ItemType.Ore] = 1;
+        _bonusProduction[ItemType.Power] = 1;
+    }
+
+   
+
+    /// <summary>
+    /// Adds the selectd customisation of type RoboticonCustomisations to the Roboticon.
+    /// Adds selected customisation's resource multiplier to the selected bonus 
+    /// </summary>
+    /// <param name="selectedCustomisation"> An instanciated class of RoboticonCustomisation</param>
+    public void CustomiseRoboticon(RoboticonCustomisation selectedCustomisation)

```

Comment:
This method should only be able to be called by the class responsible for the customisations of roboticons so that we can enforce the transaction happening properly. To achieve this we might have to move the roboticon class into the roboticon factory class (or whatever we call the class in charge of customisation transactions) and making roboticon a private member of that class but accessible through it